server/src/main/java/org/opensearch/repositories/RepositoriesService.java

```diff
@@ -699,6 +707,41 @@ public static void validateRepositoryMetadataSettings(
                     + minVersionInCluster
             );
         }
+        if (SHALLOW_SNAPSHOT_V2.get(repositoryMetadataSettings)) {
+            if (minVersionInCluster.onOrAfter(Version.V_2_17_0) == false) {
+                throw new RepositoryException(
+                    repositoryName,
+                    "setting "
+                        + SHALLOW_SNAPSHOT_V2.getKey()
+                        + " cannot be enabled as some of the nodes in cluster are on version older than "
+                        + Version.V_2_17_0
+                        + ". Minimum node version in cluster is: "
+                        + minVersionInCluster
+                );
+            }
+            if (repositoryWithShallowV2Exists(repositories)) {
```

Comment:
Will this cause any issues for users who may have registered multiple repositories with shallow snapshot v2?

Reply:
We need to add this change in 2.17 version, otherwise we can't add this restriction in 2.18